octopusdeploy/templates/pvc.yaml

```diff
@@ -0,0 +1,41 @@
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+  name: repository-claim
+spec:
+  # If you are using one replica, ensure the accessMode value is -ReadWriteOnce
+  # If you have more than one replica, ensure the accessMode value is -ReadWriteMany
+  accessModes:
+    - ReadWriteMany
+  storageClassName: {{ .Values.storageClassName }}
```

Comment:
This change will mean that the `storageClassName` is always defined, even if it is an empty string.

The blog post at https://itnext.io/kubernetes-tip-how-to-use-the-storageclass-attribute-75cf47e7c6b0 has some details on the difference between not including the field, including an empty string, or defining a value.

Reply:
Hey @mcasperson,

I could totally be wrong here, but..

From what I'm seeing in the original code, it was either `""` (an empty string, as you said) or there was a value. This was done via the if/else logic. Which means there was a value either way, either `""` or a value that a user set.

With the code I implemented, can't it be done the same way? For example, a user can define `""` or a real value in the `.Values.storageClassName`, which would effectively make the outcome the same.

Per the blog post you linked:
`In this case, just set storageClass to an empty string ("") in the PersistentVolumeClaim. This will suppress dynamic provisioning!`